storage/innobase/include/ha_prototypes.h

```diff
@@ -462,5 +462,15 @@ void destroy_background_thd(MYSQL_THD thd);
 void
 innobase_reset_background_thd(MYSQL_THD);
 
+#ifdef WITH_WSREP
+/** Append table-level exclusive key.
+@param thd   MySQL thread handle
+@param table table
+@retval false on success
+@retval true on failure */
+struct dict_table_t;
+bool wsrep_append_table(MYSQL_THD thd, const dict_table_t &table);
+#endif /* WITH_WSREP */
+
```

Comment:
why declaration in ha_prototypes.h? note that wsrep_append_keys() has declaration in ha_innodb.h

Reply:
Because that function is not part of any class, ha_innobase::wsrep_append_keys is part of ha_innobase class and used only in innobase/handler directory. I would not include ha_innodb.h in lower levels if not necessary.